projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -539,6 +547,22 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this.zone.runOutsideAngular(() => this.document.defaultView.removeEventListener('resize', this.resizeHandler));
         this.destroy$.next(true);
         this.destroy$.complete();
+
+        if (this._subscribeOnRowAdded$) {
```

Comment:
The `unsubscribe` should be in the `onDestroy` hook. Use the approach in the previous comment and this whole code snippet is redundant.